release/release_utils.py

```diff
@@ -390,21 +399,26 @@ def hg_push(repo_root):
     else:
         execute('hg -R %s push' % repo_root)
 
-# Pull the latest changes and update
-def update_project(path):
-    if git_bare_repo_exists_at_path(path):
-        execute('git -C %s fetch' % path)
-    elif is_git(path):
-        execute('git -C %s pull' % path)
+# Clean repo, pull the latest changes and update
```

Comment:
The intended meaning of a "clean repo" is a repo that cannot be distinguished from a freshly cloned repo. This is what the script was previously doing - discarding the repos altogether and making a fresh clone each time (without a warning the user could respond to when the --auto option was provided).
The "Verify repositories" step will fail if there are discrepancies, so I'll change the "Clone/update repositories" step so that it doesn't discard changes, allowing the verify step to just fail.